File: tutorial5-Restful-services-integration/tutorial5-Restful-services-integration/java/hacker-news-consuming-rest-example/src/main/java/com/example/hnconsumingrest/MyHackerNewsController.java
package com.example.hnconsumingrest;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.client.RestTemplate;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

// TODO: Class Annotations
public class MyHackerNewsController {
    private static final Logger logger = LoggerFactory.getLogger(MyHackerNewsController.class);

    // TODO: Method Annotations
    private List<HackerNewsItemRecord> hackerNewsTopStories(/* TODO: RequestParams for search */) {
        // TODO: Get IDs of top stories

        // TODO: Iterate through each ID and check if it contains the "search" terms

        return null;
    }
}

File: tutorial5-Restful-services-integration/tutorial5-Restful-services-integration/python/hacker-news-consuming-rest-example/src/MyHackerNewsController.py
from fastapi import APIRouter, Query
from typing import List, Optional
import requests
import logging
from pydantic import BaseModel

router = APIRouter()
logging.basicConfig(level=logging.INFO)
logger = logging.getLogger("hn")

# Pydantic models
class HackerNewsItemRecord(BaseModel):
    id: Optional[int]
    title: Optional[str]
    by: Optional[str]
    url: Optional[str]
    time: Optional[int]
    score: Optional[int]
    type: Optional[str]
    descendants: Optional[int]
    kids: Optional[List[int]]
    text: Optional[str]

#TODO: method annotations 
def hacker_news_top_stories(search: Optional[str] = Query(None)):
    #TODO: Get IDs of top stories

    #TODO: Iterate through each ID and check if it contains the "search" terms

    return ""
